message-core/src/main/java/net/spals/appbuilder/message/core/blockingqueue/BlockingQueueMessageConsumerPlugin.java

```diff
@@ -60,8 +59,8 @@
         this.blockingMessageQueue = blockingMessageQueue;
         // The number of registered consumer callbacks provides an upper bound on
         // the number of executor threads that we'll need.
-        this.executorService = executorServiceRegistry.registerExecutorService(getClass(),
-                Executors.newFixedThreadPool(Math.max(consumerCallbackSet.size(), 1)));
+        this.executorService = executorServiceFactory.createFixedThreadPool(Math.max(consumerCallbackSet.size(), 1),
+                getClass());
```

Comment:
Any `ExecutorService` created by the `ExecutorServiceFactory` is automatically attached to the application lifecycle. This means that the `ExecutorService` will run graceful shutdown code when the application is shutdown.